pkg/format/sarif.go

```diff
@@ -62,11 +62,11 @@ type ArtifactLocation struct {
 type Region struct {
 	StartLine   int          `json:"startLine,omitempty"`
 	StartColumn int          `json:"startColumn,omitempty"`
-	EndLine     int          `json:"EndLine,omitempty"`
-	EndColumn   int          `json:"EndColumn,omitempty"`
-	ByteOffset  int          `json:"ByteOffset,omitempty"`
-	ByteLength  int          `json:"ByteLength,omitempty"`
-	Snippet     SnippetSarif `json:"snippet"`
+	EndLine     int          `json:"endLine,omitempty"`
+	EndColumn   int          `json:"endColumn,omitempty"`
+	ByteOffset  int          `json:"byteOffset,omitempty"`
+	ByteLength  int          `json:"byteLength,omitempty"`
+	Snippet     SnippetSarif `json:"snippet,omitempty"`
```

Comment:
This is not a good way of omitting a struct. omitempty will work only for types that have well-defined null values, so natural types, arrays and pointers. To properly omit, change SnippetSarif to *SnippetSarif.